### ISSUE TITLE
Remove unneeded files

### ISSUE THREAD

**Issue by PencilShavings** [AngelinaMinshall/amdesignsportfolio#2]
This issue proposes to remove unneeded files and how to prevent them from coming back.

## Remove .DS_Store
These are stupid files that macOS Finder likes to put in places.
Any file that starts with a `.` is hidden in Finder and cannot be seen unless it is told to do so.
While they don't hurt anything they clutter things up _(Personal opinion)._

Two DS_Store files can be found here.
- `amdesignsportfolio/.DS_Store`
- `amdesignsportfolio/webportfolio/.DS_Store`

## Remove node_modules
The `node_modules` folder and all of its contents are created when Node ( `npm`) installs packages.
While these packages could be relevant to to your project, they are not your project.
Each library in your `node_packages` folder are their own project in their own right with their own issues.
You don't need to be responsible for maintaining other library's or frameworks.
The `package.json` & `package-lock.json` should contain all package names and versions need to retrieve the files when you are ready to deploy.

A redeployment of the project would look like so.
```
git clone https://github.com/AngelinaMinshall/amdesignsportfolio.git
cd amdesignsportfolio`
npm install
# Open index.html in web browser
```
TL;DR: The  `node_modules` may need to be on your computer for things to function but they don't have to be in your version control!

## Solution
In your project directory you can run the following in the command line.
_An equivalent can be done in VSCode but I don't use VSCode when working with git, so call me old school_
```
git rm --cached .DS_Store
git rm --cached webportfolio/.DS_Store
git rm --cached -r node_modules
git commit -m "Removed node_modules & .DS_Store"
git push
```
They should now be removed from the git project but still be on your computer.
Run a `git status` to see.

## Add a `.gitignore`
A gitignore file is a hidden file that will exclude files and folders from git. It is useful for excluding unwanted stuff from entering your git project.
Create a `.gitignore` file in the root of your project and add the following:
```
*.DS_Store
node_modules
```
This will make it so git will no longer see any `.DS_Store` files or the `node_modules` folder.
Run a `git status` to see.

Commit `.gitignore` to the project
```
git add .gitignore
git status
git commit -m "Added .gitignore"
git push

```

_NOTE: When committing anything make sure you only have changes related to what you are working on. Otherwise you may commit changes unrelated to the commit message or commit partial worked on code that could break something._
